students/itsabhi/Lesson5/mailroom3.py

```diff
@@ -0,0 +1,88 @@
+#! /usr/bin/env python3
+
+
+def sending_thanksv2(donor_hist):
+    # If already exists, add donation amt to history, else update records as a new donor & then print thank you letter.
+    donor = input("Type the Full Name of the donor here or type List to display current donors : ")
+    if donor == "list":
+                for k in donor_hist:
+                    print(k)
+                donor = input("Type the Full Name of the donor here: ")
+    if donor in donor_hist.keys():
+        while True:
+            amount = input("Enter Donation value: ")
+            donation_hist = donor_hist.get(donor)
+            try:
+                donor_hist[donor] = (donation_hist + [int(amount)])
+            except ValueError:
+                print("Invalid donation entered..")
+            else:
+                break
+    else:
+        while True:
+            amount = input("Enter Donation value: ")
```

Comment:
it would be better to extract this and lines 12-13 and do try / except to validate user input then do add or append